api/include/opentelemetry/trace/global_provider.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <mutex>
+
+#include "opentelemetry/trace/noop.h"
+#include "opentelemetry/trace/tracer_provider.h"
+#include "opentelemetry/nostd/unique_ptr.h"
+
+using opentelemetry::trace::NoopTracer;
+using opentelemetry::trace::TracerProvider;
+
+std::mutex provider_mutex;
+
+namespace opentelemetry
+{
+
+class DefaultTracerProvider final : public TracerProvider
```

Comment:
Should this class should be somehow hidden from consumers of otel? I don't think users should be creating instances of it.

Maybe make the constructor private and `friend class Provider`?